infra/modules/azure_function_app/outputs.tf

```diff
@@ -0,0 +1,30 @@
+output "subnet" {
+  value = {
+    id   = azurerm_subnet.this.id
+    name = azurerm_subnet.this.name
+  }
+}
+
+output "storage_account" {
+  value = {
+    id   = azurerm_storage_account.this.id
+    name = azurerm_storage_account.this.name
+  }
+}
+
+output "function_app" {
+  value = {
```

Comment:
If you agree, I would add the id of the function app's identity. With this it would be easy to add additional role assignments in the root module

Reply:
Nice catch